packages/slate-editor/src/modules/editor/createEditor.ts

```diff
@@ -25,7 +22,7 @@ import {
 } from './plugins';
 
 export function createEditor(
-    baseEditor: Editor,
+    baseEditor: SlateEditor,
     getExtensions: () => Extension[],
```

Comment:
Maybe not a question for this round of refactoring, but do you know why we pass the extensions as a getter callback? Why not pass it directly as an array?

Reply:
DEV-19122

Reply:
Judging from [this code](https://github.com/prezly/slate/blob/main/packages/slate-editor/src/modules/editor/useCreateEditor.ts#L43-L46), it seems like it's just there so the `editor` is not recreated when/if extensions array changes? 🤔